app/controllers/api/items_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::ItemsController < Api::BaseController
+  def fetch_latest_products
+    page = params[:page].to_i
+    limit = params[:limit].to_i
+    offset = (page - 1) * limit
```

Comment:
Potential security concern: The `page` and `limit` parameters are directly converted to integers without validation. This could lead to unexpected behavior if non-integer values are passed. Consider adding validation to ensure these parameters are valid integers.